docs/dcp2_system_design.rst

```diff
@@ -983,6 +1062,398 @@ running the import tool with be requested manually in
+   the staging area will be imported into. 
+
+   The ``entity_id`` and ``version`` fields in the name of the descriptor must
+   match those in the name of the object created in the next step.
+
+3. Add the entity describing the data file as defined in `Add an entity_`.
+
+This type of update creates an orphaned data file so it should only be used as
+part of other updates, as described in `Add a subgraph`_, `Add a project`_ or
+`Add an entity to existing subgraphs`_.
+
+
+Add a subgraph
+~~~~~~~~~~~~~~
+
+1. Add any entities introduced by the subgraph as described in `Add an
+   entity`_. If the subgraph also refers to entities that were previously
+   imported, those entities should not be written to the staging area.
+
+2. Write the subgraph to an object in the ``links`` directory of the staging
+   area. The ``links_id`` field in the name of that object must be different
+   from the value of the ``links_id`` column in any row of the ``links`` table
+   of the TDR dataset the staging are is imported into.
+
+
+Add a project
+~~~~~~~~~~~~~
+
+1. Add the ``project`` entity as described in `Add an entity`_   
+
+2. Add all other entities belonging to the project as described in `Add an
+   entity`_ 
+
+3. Add the subgraphs belonging to the project as described in `Add a
+   subgraph`_
+
+
+Add an entity to existing subgraphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Add the entity as described in `Add an entity`_
+
+2. Update the subgraphs that reference the entity as described in `Update a
+   subgraph`_
+
+
+Add a data file to existing subgraphs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Add the data file as described in `Add a data file`_
+
+2. Update the subgraphs that reference the data file as described in `Update a
+   subgraph`_
+
+
+Remove an entity
+~~~~~~~~~~~~~~~~
+
+1. Make sure the staging area property ``is_delta`` is ``true``.
+
+2. Create an empty object in the ``metadata`` directory of the staging area.
+   The ``entity_type`` and ``entity_id`` fields in the object name must be the
+   same as those in the name of the original (the entity to be deleted), the
+   ``version`` field must be strictly higher and the name must end in
+   ``.json.remove`` instead of just ``.json``.
+
+3. If the removed entity is of type ``project``, remove all subgraphs in that
+   project. Otherwise, update the subgraphs referencing the removed entity as
+   described in `Update a subgraph`_
+
+A removed entity must remain in all snapshots that are part of already
+published data releases but must be absent from any subsequently created
+snapshot in the data release currently being prepared.
+
+|nn| This means that the corresponding row cannot be removed from the TDR
+dataset, but instead has to be soft-deleted, so it is not included in
+subsequently created snapshots. This is a TDR implementation detail. |ne|
+
+
+Remove a data file
+~~~~~~~~~~~~~~~~~~
+
+1. Remove the entity describing the data file as described in `Remove an
```

Comment:
In our TDR model, this will manifest as a soft-delete of the row in the corresponding "_file" table (i.e., "analysis_file"), is that right?

Reply:
Yes.

Reply:
Well, not necessarily. Any mechanism that enures that the row is excluded from the next snapshot would work.

Reply:
Addressed